### PR TITLE
feat: Add vault config parameters

### PR DESCRIPTION
**Pull request by jffernandez**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does
[ ] Add a screenshot if your PR has UI changes

### CODE REVIEW

File: src/state/vaults/fetchVaults.ts [hunk 1]
```diff
@@ -83,6 +83,10 @@ const fetchVaults = async () => {
 
       const allocPoint = new BigNumber(info.allocPoint._hex);
       const poolWeight = allocPoint.div(new BigNumber(totalAllocPoint));
+      const state =
+        info.retired == true ? 'retired' : info.paused == true ? 'paused' : info.lastRewardBlock > 0 ? 'active' : 'new';
+      const proposedStrat = info.proposedStrat as string;
+      const proposedTime = info.proposedTime as Number;
```
Comment (jffernandez) [inline src/state/vaults/fetchVaults.ts:89]: When proposedTime is > 0, there is a new strategy ready to replace the working one, but there is a 24h timelock. This time is when the change was submitted (blockchain UTC time). Maybe we can implement a counter so the user knows when the new strategy will go live, and have a link to it so can check the smart contract (proposedStrat).

File: src/state/vaults/fetchVaults.ts [hunk 2]
```diff
@@ -83,6 +83,10 @@ const fetchVaults = async () => {
 
       const allocPoint = new BigNumber(info.allocPoint._hex);
       const poolWeight = allocPoint.div(new BigNumber(totalAllocPoint));
+      const state =
```
[the remark this answers is not in the document]
Comment (FranRom) [reply src/state/vaults/fetchVaults.ts:86]: Yes I think we have some example with typed states in the project

File: src/state/vaults/fetchVaults.ts (hunk 1 above)
Comment (FranRom) [reply src/state/vaults/fetchVaults.ts:89]: Can you give me an example of a newly added strategy in terms of a user?

Comment (FranRom) [reply src/state/vaults/fetchVaults.ts:89]: A new Vault added would count like that?

Comment (FranRom) [reply src/state/vaults/fetchVaults.ts:89]: ok Pancake migration is a good example

Comment (jffernandez) [reply src/state/vaults/fetchVaults.ts:89]: Every vault has a 'new' stage, starting with the configuration of the vault (while it's tested before launch) and ending with the activatios of the vault from the Gnosis Safe. From this point, it is in 'active' state, it starts to reward BALLE and must be on the app.

File: src/utils/web3.ts [hunk 3]
```diff
@@ -3,7 +3,10 @@ import { HttpProviderOptions } from 'web3-core-helpers';
 import getRpcUrl from 'utils/getRpcUrl';
 
 const RPC_URL = getRpcUrl();
-const httpProvider = new Web3.providers.HttpProvider(RPC_URL, { timeout: 10000 } as HttpProviderOptions);
+const FALLBACK_RPC_URL = 'https://data-seed-prebsc-1-s1.binance.org:8545/';
```
Comment (jffernandez) [inline src/utils/web3.ts:6]: The config system should be revised to avoid this hack.

File: src/config/constants/types.ts [hunk 4]
```diff
@@ -16,13 +16,24 @@ export interface Token {
   projectLink?: string;
 }
 
+export type VaultType = 'autoCompound' | 'balle';
+
+export type DepositType = 'LP' | 'singleAsset';
+
+export type VaultState = 'new' | 'active' | 'paused' | 'retired';
```
Comment (FranRom) [inline src/config/constants/types.ts:23]: Maybe those should be Enum instead of types, WDYT?

Comment (jffernandez) [reply src/config/constants/types.ts:23]: Yes, if you like more can be changed. I usually use types for typing.

**Review by FranRom — APPROVED** [review]
Let's go!

**Pull request merged**